Storage/Storage/Extensions/NSManagedObjectContext+Storage.swift

```diff
@@ -61,7 +60,11 @@ extension NSManagedObjectContext: StorageType {
     ///
     public func deleteObject<T: Object>(_ object: T) {
         guard let object = object as? NSManagedObject else {
-            logErrorAndExit("Invalid Object Kind")
+            let errorMessage = "deleteObject failed. Invalid Object Kind \(type(of: object)) " +
+                               "for object type: \(T.self). This error is related to NSManagedObject " +
+                               "and CoreData."
+            DDLogError(errorMessage)
```

Comment:
I've noticed that in this and other changes, we no longer use `logErrorAndExit`; we just log the error without exiting the execution with a fatalError. Is this intentional? I'm concerned that this might leave the execution in a corrupted state, potentially causing unpredictable issues.